### PR TITLE
Fix bug where parenthesized lambdas were being classified as top level

### PR DESCRIPTION
**Pull request by davidwengier**
Fixes https://github.com/dotnet/roslyn/issues/51297

Missed one type of lambda when combining the two syntax comparers.

### CODE REVIEW

**Review by tmat — APPROVED** [review]
:shipit:

File: src/EditorFeatures/CSharpTest/EditAndContinue/TopLevelEditingTests.cs [hunk 1]
```diff
@@ -10161,6 +10186,34 @@ class C
                 Diagnostic(RudeEditKind.NotCapturingVariable, "a", "a"));
         }
 
+        [Fact, WorkItem(51297, "https://github.com/dotnet/roslyn/issues/51297")]
+        public void IndexerWithExpressionBody_Update_LiftedParameter_2()
```
Comment (tmat) [inline src/EditorFeatures/CSharpTest/EditAndContinue/TopLevelEditingTests.cs:10190]: >IndexerWithExpressionBody_Update_LiftedParameter_2 [](start = 20, length = 50)

Let's add a variant of this test that uses local function and old-style lambda delegate (unless there is already one somewhere else)

**Comment by tmat** [comment]
>             if (_compareStatementSyntax)

Let's explain here why these differ and what criteria do we use to label the same kinds differently.

---
Refers to: src/Features/CSharp/Portable/EditAndContinue/SyntaxComparer.cs:739 in 54658ab. [](commit_id = 54658ab4f656df00ca72b73f3cb39f64835d0485, deletion_comment = False)

**Comment by davidwengier** [comment]
Nope, didn't just miss one type. Need to change my approach here.

I was trying to keep the classification all in one switch statement as much as possible, for simplicity and hoping it might help for future work with top the level statements feature, but because anonymous methods and block-bodied lambdas exist, that approach can't work. Something like defining a property as `int X => new Func(() => { /* any number and kind of C# statements */ })();` means we have the top level comparer potentially seeing any statement syntax node, and wanting it to be not be labelled.

So I need to move way more of the classification to apply only for statement syntax. Thankfully thats not too hard.

**Comment by tmat** [comment]
Makes sense. I thought we might need to do that.

**Pull request merged**